### PR TITLE
Chunk pool format asyn

### PR DESCRIPTION
**Pull request by Vigor-jpg**
### What problem does this PR solve?

Format chunk file pool asyn when we deploy the cluster.

Issue Number: #xxx 

Problem Summary: This is just a draft, I do some core work for this task but it is not complete.

### What is changed and how it works?

What's Changed:  

How it Works: 

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [x] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by Cyber-SiKu** [comment]
@Vigor-jpg please resolve the conflict

File: src/chunkserver/chunkserver.cpp [hunk 1]
```diff
@@ -65,6 +65,7 @@ DEFINE_string(raftSnapshotUri, "curve://./0/copysets", "raft snapshot uri");
 DEFINE_string(raftLogUri, "curve://./0/copysets", "raft log uri");
 DEFINE_string(recycleUri, "local://./0/recycler" , "recycle uri");
 DEFINE_string(chunkFilePoolDir, "./0/", "chunk file pool location");
+DEFINE_uint32(chunkFillPoolPercent, 10, "format percent for chunkfillpool");
```
Comment (xu-chaojie) [inline src/chunkserver/chunkserver.cpp:68]: use size instead percent of disk will be better

File: src/chunkserver/datastore/file_pool.cpp [hunk 2]
```diff
@@ -217,18 +218,30 @@ FilePool::FilePool(std::shared_ptr<LocalFileSystem> fsptr)
...
+    //LOG(ERROR) << "fileSize:" << poolOpt_.fileSize;
+    //LOG(ERROR) << "preAllocateNum:" << poolOpt_.preAllocateNum;
     if (poolOpt_.getFileFromPool) {
-        if (!CheckValid()) {
-            LOG(ERROR) << "check valid failed!";
-            return false;
+        currentdir_ = poolOpt_.filePoolDir;
+        currentState_.chunkSize = poolOpt_.fileSize;
+        currentState_.metaPageSize = poolOpt_.metaPageSize;
+        if (!fsptr_->DirExists(currentdir_.c_str())) {
```
Comment (xu-chaojie) [inline src/chunkserver/datastore/file_pool.cpp:228]: First check whether the existing chunkfile is legal, and then format the newly added chunkfile in the sub-thread

File: src/chunkserver/datastore/file_pool.cpp [hunk 3]
```diff
@@ -356,6 +369,134 @@ void FilePool::CleanWorker() {
...
+
+int FilePool::FormatWorker() {
+    std::vector<Thread> threads;
+    std::vector<bool> chunkNums;
+    chunkNums.resize(poolOpt_.preAllocateNum, false);
+    uint64_t chunkMaxIndex = 0;
+    LOG(INFO) << "format work start!";
+
+    // check file num order.
+    {
```
Comment (xu-chaojie) [inline src/chunkserver/datastore/file_pool.cpp:414]: You can check this in the main thread

**Comment by wuhongsong** [comment]
@Vigor-jpg  continue?

**Comment by Vigor-jpg** [comment]
> @Vigor-jpg continue?

Yes, I am making a change to this.

**Pull request closed without merging**